docs/react/options.md

```diff
@@ -0,0 +1,136 @@
+# Setting options
+
+## Configure with option components
+
+The Highcharts React integration includes customizable components such as
+Title, Subtitle, Credits, Tooltip, Legend, XAxis, YAxis, and PlotOptions,
+providing flexible options to configure and fine-tune your charts using JSX.
+
+Enhance your chart by adding option components:
+
+```jsx
+import {
+    Title,
+    Subtitle,
+    Credits,
+    XAxis,
+    YAxis,
+    PlotOptions,
+    Tooltip,
+    Legend
+} from '@highcharts/react';
+```
+
+Here's an example of adding a custom title:
+
+```jsx
+<Chart>
+  <Title>Custom Chart Title</Title>
+  <Series type="column" data={[1, 2, 3]} />
+</Chart>
+```
+
+Most top-level properties can be set as props. If an option component has a `text` or `format` property,
+you can set it by passing children to the component. In most cases, a component that accepts strings as children will be formatted
+as a [format string in Highcharts](https://www.highcharts.com/docs/chart-concepts/labels-and-string-formatting#format-strings).
+
+With the `Tooltip` component, you can customise the information displayed using a [format string](https://www.highcharts.com/docs/chart-concepts/templating). Here's a simple example:
+
+```jsx
+<Chart>
+  <Series type="column" data={[1, 2, 3]} />
+
+  <Tooltip>{'X: {point.x}, Y: {point.y}'}</Tooltip>
+</Chart>
+```
+
+If you prefer more flexibility, you can use React components and elements as children:
+
+```jsx
+function TooltipFormat() {
+  return (
+    <>
+      <div data-hc-option="headerFormat">
```

Comment:
Could we add an example of a "normal", standalone React component without mapping with `data-hc-option` as well? Perhaps with some reactive prop?